lib/mirage.mli

```diff
@@ -461,7 +461,15 @@ val conduit_direct: ?tls:bool -> stackv4 impl -> conduit impl
 
 type http
 val http: http typ
+
 val http_server: conduit impl -> http impl
+[@@ocaml.deprecated "`http_server` is deprecated. Please use `cohttp_server` or `httpaf_server` instead."]
```

Comment:
Instead of deprecating this, how about just making it a choice between Cohttp and Httpaf via an optional argument, and then swapping the defaults in a future release as httpaf support matures?  Most users will probably not care too much about the specific server implementation chosen, and advanced users have the functions below with the specific choice.

Reply:
since there is no common signature of httpaf and cohttp yet (esp. in respect to headers etc.), I don't think application developers can be http-server agnostic right now. in the future, this will hopefully be cleared up.